File: Branches/VBInicial/vpPriV100GrupoMundifios/CopiaEntreEmpresas/Vendas/FichaTabDocVendas/VndIsFichaTabDocVendas.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports Primavera.Extensibility.Sales.Editors
Imports Primavera.Extensibility.BusinessEntities
Imports Primavera.Extensibility.BusinessEntities.ExtensibilityService.EventArgs
Imports Vimaponto.PrimaveraV100.Clientes.GrupoMundifios.Generico

Namespace CopiaEntreEmpresas
    Public Class VndIsFichaTabDocVendas
        Inherits FichaTabDocVendas

        Public Overrides Sub AntesDeGravar(ByRef Cancel As Boolean, e As ExtensibilityEventArgs)
            MyBase.AntesDeGravar(Cancel, e)
            If Module1.VerificaToken("CopiaEntreEmpresas") = 1 Then

                If Not ValidacaoCamposUtilVenda Then Cancel = True

                If Cancel = False Then If Not ValidacaoCamposUtilCompra() Then Cancel = True

            End If

        End Sub


        '#edusamp
        Private Function ValidacaoCamposUtilVenda() As Boolean

            On Error GoTo TrataErro

            Dim DocVendaDestino As String
            Dim SerieVendaDestino As String

            Me.Documento.CamposUtil("CDU_TipoDocVendasDestino").Valor = UCase(Me.Documento.CamposUtil("CDU_TipoDocVendasDestino").Valor & "")

            Me.Documento.CamposUtil("CDU_SerieVendasDestino").Valor = UCase(Me.Documento.CamposUtil("CDU_SerieVendasDestino").Valor & "")

            DocVendaDestino = UCase(Me.Documento.CamposUtil("CDU_TipoDocVendasDestino").Valor & "")

            'Se campo da Base de dados vazia não avisa nada e sai
            If Len(DocVendaDestino) > 0 Then

                If Me.Documento.TipoDocumento <> BSO.Vendas.TabVendas.DaValorAtributo(UCase(Me.Documento.CamposUtil("CDU_TipoDocVendasDestino").Valor & ""), "TipoDocumento") Then
                    MsgBox("O tipo de documento de Venda configurado não é permitido.", vbCritical, "Campos de utilizador Doc. Venda incompletos")
                    ValidacaoCamposUtilVenda = False
                    Exit Function
                End If

                SerieVendaDestino = UCase(Me.Documento.CamposUtil("CDU_SerieVendasDestino").Valor & "")

                If Len(SerieVendaDestino) > 0 Then

                    ValidacaoCamposUtilVenda = True
                    Exit Function

                Else

                    MsgBox("Série não preenchida para o Documento de Venda " & DocVendaDestino & ".", vbCritical, "Campos de utilizador Doc. Venda incompletos")
                    ValidacaoCamposUtilVenda = False
                    Exit Function
                End If


            Else

                'Se não tiver documento preenchido, apago a serie caso exista
                Me.Documento.CamposUtil("CDU_SerieVendasDestino").Valor = ""
                ValidacaoCamposUtilVenda = True
                Exit Function
            End If


TrataErro:
            MsgBox("Erro: " & Err.Description, vbCritical, "Erro nos campos de utilizador Doc. Venda ")
            ValidacaoCamposUtilVenda = False

        End Function


        '#edusamp
        Private Function ValidacaoCamposUtilCompra() As Boolean

            On Error GoTo TrataErro

            Dim DocCompraDestino As String
            Dim SerieCompraDestino As String

            Me.Documento.CamposUtil("CDU_TipoDocComprasDestino").Valor = UCase(Me.Documento.CamposUtil("CDU_TipoDocComprasDestino").Valor & "")
            Me.Documento.CamposUtil("CDU_SerieComprasDestino").Valor = UCase(Me.Documento.CamposUtil("CDU_SerieComprasDestino").Valor & "")

            DocCompraDestino = UCase(Me.Documento.CamposUtil("CDU_TipoDocComprasDestino").Valor & "")

            'Se campo da Base de dados vazia não avisa nada e sai
            If Len(DocCompraDestino) > 0 Then

                If Me.Documento.TipoDocumento <> BSO.Compras.TabCompras.DaValorAtributo(UCase(Me.Documento.CamposUtil("CDU_TipoDocComprasDestino").Valor & ""), "TipoDocumento") Then
                    MsgBox("O tipo de documento de Compra configurado não é permitido.", vbCritical, "Campos de utilizador Doc. Venda incompletos")
                    ValidacaoCamposUtilCompra = False
                    Exit Function
                End If

                SerieCompraDestino = UCase(Me.Documento.CamposUtil("CDU_SerieComprasDestino").Valor & "")

                If Len(SerieCompraDestino) > 0 Then

                    ValidacaoCamposUtilCompra = True
                    Exit Function

                Else

                    MsgBox("Série não preenchida para o Documento de Compra " & DocCompraDestino & ".", vbCritical, "Campos de utilizador Doc. Compra incompletos")
                    ValidacaoCamposUtilCompra = False
                    Exit Function
                End If

            Else

                'Se não tiver documento preenchido, apago a serie caso exista
                Me.Documento.CamposUtil("CDU_SerieComprasDestino").Valor = ""
                ValidacaoCamposUtilCompra = True
                Exit Function
            End If

TrataErro:
            MsgBox("Erro: " & Err.Description, vbCritical, "Erro nos campos de utilizador Doc. Compra ")
            ValidacaoCamposUtilCompra = False

        End Function



    End Class
End Namespace

File: Trunk/vpPriV100GrupoMundifios/CopiaEntreEmpresas/Vendas/FichaTabDocVendas/VndIsFichaTabDocVendas.cs
using Generico;
using Microsoft.VisualBasic;
using Primavera.Extensibility.BusinessEntities.ExtensibilityService.EventArgs;
using Primavera.Extensibility.Sales.Editors;
using System.Windows.Forms;

namespace CopiaEntreEmpresas
{
    public class VndIsFichaTabDocVendas : FichaTabDocVendas
    {
        public override void AntesDeGravar(ref bool Cancel, ExtensibilityEventArgs e)
        {
            base.AntesDeGravar(ref Cancel, e);
            if (Module1.VerificaToken("CopiaEntreEmpresas") == 1)
            {
                if (!ValidacaoCamposUtilVenda())
                    Cancel = true;

                if (Cancel == false)
                {
                    if (!ValidacaoCamposUtilCompra())
                        Cancel = true;
                }
            }
        }

        private bool ValidacaoCamposUtilVenda()
        {
            string DocVendaDestino;
            string SerieVendaDestino;
            try
            {
                this.Documento.CamposUtil["CDU_TipoDocVendasDestino"].Valor = Strings.UCase(this.Documento.CamposUtil["CDU_TipoDocVendasDestino"].Valor + "");

                this.Documento.CamposUtil["CDU_SerieVendasDestino"].Valor = Strings.UCase(this.Documento.CamposUtil["CDU_SerieVendasDestino"].Valor + "");

                DocVendaDestino = Strings.UCase(this.Documento.CamposUtil["CDU_TipoDocVendasDestino"].Valor + "");

                if (Strings.Len(DocVendaDestino) > 0)
                {
                    if (Documento.TipoDocumento != BSO.Vendas.TabVendas.DaValorAtributo(Strings.UCase(Documento.CamposUtil["CDU_TipoDocVendasDestino"].Valor + ""), "TipoDocumento"))
                    {
                        MessageBox.Show("O tipo de documento de Venda configurado não é permitido.", "", MessageBoxButtons.OK, MessageBoxIcon.Information);
                        return false;
                    }

                    SerieVendaDestino = Strings.UCase(Documento.CamposUtil["CDU_SerieVendasDestino"].Valor + "");

                    if (Strings.Len(SerieVendaDestino) > 0)
                        return true;
                    else
                    {
                        MessageBox.Show("Série não preenchida para o Documento de Venda " + DocVendaDestino + "." + "Campos de utilizador Doc. Venda incompletos", "", MessageBoxButtons.OK, MessageBoxIcon.Information);

                        return false;
                    }
                }
                else
                {
                    Documento.CamposUtil["CDU_SerieVendasDestino"].Valor = "";
                    return true;
                }
            }
            catch
            {
                MessageBox.Show("Erro nos campos de utilizador Doc. Venda", "", MessageBoxButtons.OK, MessageBoxIcon.Error);
                return false;
            }
        }

        private bool ValidacaoCamposUtilCompra()
        {
            try
            {
                string DocCompraDestino;
                string SerieCompraDestino;

                this.Documento.CamposUtil["CDU_TipoDocComprasDestino"].Valor = Strings.UCase(this.Documento.CamposUtil["CDU_TipoDocComprasDestino"].Valor + "");
                this.Documento.CamposUtil["CDU_SerieComprasDestino"].Valor = Strings.UCase(this.Documento.CamposUtil["CDU_SerieComprasDestino"].Valor + "");

                DocCompraDestino = Strings.UCase(this.Documento.CamposUtil["CDU_TipoDocComprasDestino"].Valor + "");

                if (Strings.Len(DocCompraDestino) > 0)
                {
                    if (Documento.TipoDocumento != BSO.Compras.TabCompras.DaValorAtributo(Strings.UCase(Documento.CamposUtil["CDU_TipoDocComprasDestino"].Valor + ""), "TipoDocumento"))
                    {
                        MessageBox.Show("O tipo de documento de Compra configurado não é permitido.", "", MessageBoxButtons.OK, MessageBoxIcon.Error);

                        return false;
                    }

                    SerieCompraDestino = Strings.UCase(Documento.CamposUtil["CDU_SerieComprasDestino"].Valor + "");

                    if (Strings.Len(SerieCompraDestino) > 0)
                        return true;
                    else
                    {
                        MessageBox.Show("Série não preenchida para o Documento de Compra " + DocCompraDestino, "", MessageBoxButtons.OK, MessageBoxIcon.Error);
                        return false;
                    }
                }
                else
                {
                    Documento.CamposUtil["CDU_SerieComprasDestino"].Valor = "";
                    return true;
                }
            }
            catch
            {
                MessageBox.Show("Erro nos campos de utilizador Doc. Compra ", "", MessageBoxButtons.OK, MessageBoxIcon.Error);
                return false;
            }
        }
    }
}